cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/io/cdap/cdap/etl/api/connector/SampleRequest.java

```diff
@@ -30,11 +30,17 @@ public class SampleRequest {
   private final String path;
   private final int limit;
   private final Map<String, String> properties;
+  private final Long timeoutMs;
 
-  private SampleRequest(@Nullable String path, int limit, Map<String, String> properties) {
+  private SampleRequest(@Nullable String path, int limit, Map<String, String> properties, @Nullable Long timeoutMs) {
```

Comment:
I don't see this constructor used anywhere

Reply:
This constructor is necessary for frontend JSON -> SampleRequest conversion. I will be adding a code comment clarifying that.

Reply:
Clarification added. Thanks for the feedback!

Reply:
Make it protected or package protected then. And change your comment to a javadoc. Are both constructors needed? Also is it tested with a unit test?

Reply:
Thanks for the feedback, I have made the changes requested. Adin determined that both constructors are not needed, and has removed the second constructor. `SampleRequest` currently does not have any unit tests that I am aware of.

Reply:
I've added some tests for SampleRequest, let me know if you think they're enough